### DISCUSSION TITLE
Upload statistics?

### DISCUSSION THREAD

**Discussion by verisimillipede** [slskd/slskd#1124] [Q&A]
Hi.

First of all I want to thank you for this soulseek client. I've been able to run this on my server 24/7. With this, I am curious if there is a way to view upload statistics? I get hundreds of downloads a day and I'd love to see some stats! 

Thanks in advance.

**Comment by earwigcahootz** [comment]
I came here to see if this feature had been requested already. I would also love this feature. I tested Nicotine because they have some rudimentary stat tracking, but the UI and overall functionality don't compare so I ditched it immediately and continue to run SLSKD.

**Comment by p0lygun** [comment]
+1, i'll have some free time at the end of  month, i'll try to propose (and hopefully implement) a solution then

**Comment by hawwwwwk** [comment]
would love something like this still :)

**Comment by oregonpillow** [comment]
I just created my own slskd exporter and dashboard, feel free to try: https://github.com/oregonpillow/slskd-exporter

**Reply by verisimillipede** [reply]
This looks great, I'll give it a shot

**Comment by jpdillingham** [comment]
I'm working on a branch that implements upload and download statistics, I will get it out within a couple of weeks, along with a new dashboard in the UI.

**Reply by oregonpillow** [reply]
hi @jpdillingham , is it currently possible to get live metrics with the rest-api ? 

When i went through the swagger the most relevant metrics for downloads/uploads/speeds etc  seemed to be from `/api/v0/telemetry/reports/transfers/summary`, but this seems to be a "look back" only endpoint, since it seems to calculate averages based on the given time window. E.g. if i put start date as 1 day, then it would return values based on averages over the past 24 hours, correct?

Personally, i think a much shorter time window would be preferable e.g. calculate like 5 mins or 1min; for metric exporters like my own, when you graph in Grafana a rolling 24 hour window of averages some graphs will barely change because the deviations gets averaged out. Versus if i had a live rolling window of the past 5 mins i could clearly see "when" a particular event was happening. Does that make sense? 😅 fantastic project btw.

**Reply by jpdillingham** [reply]
[This branch](https://github.com/slskd/slskd/compare/master...reports) will give the best idea of what metrics are coming.  All of the metrics are available in prometheus format via the `/metrics` endpoint and `/telemetry/metrics` has a few endpoints that return them in JSON.  I have a `/telemetry/metrics/kpis` endpoint that I'll use to populate the new dashboard.

It sounds like an endpoint that will allow some filtering to select which metrics to return would be useful for what you're doing; I can add that when I revisit this branch.
